### PR TITLE
Run eslint and unit tests before Docker build in Jenkins, remove directory traversal

### PR DESCRIPTION
**Pull request by hhellyer**
Save the base directory to ensure we can return to it in the Jenkinsfile.

### CODE REVIEW

**Comment by james-wallis** [comment]
### Summary
* Move portal and turbine eslint and unit tests to above the Docker build. (exposed https://github.com/eclipse/codewind/issues/1673)
  * Means the PR build will fail sooner if eslint or a unit test fails so we can get faster feedback
  * Additionally we will stop running `docker build`s for Jenkins builds which are destined to fail anyway
* Ensure that all directory traversal commands in the Jenkins built use a `$DIR` variable to stop errors in the future
  * Fix for the final bullet point in https://github.com/eclipse/codewind/issues/242

### Testing
* Ensured the Jenkins build passes

File: Jenkinsfile [hunk 1]
```diff
@@ -15,6 +15,112 @@ pipeline {
...
+                        if [ $? -ne 0 ]; then
+                            exit 1
+                        fi
+
+                        npm run eslint
+                        if [ $? -ne 0 ]; then
+                            exit 1
+                        fi
+
+                        cd $DIR
```
Comment (jcockbain) [inline Jenkinsfile]: is this used now?

Comment (james-wallis) [reply Jenkinsfile]: 🥇

Comment (james-wallis) [reply Jenkinsfile]: Removed

**Pull request merged**